# Assembly
Raw read quality assessment and filtering.

Here, the data is the sequence generated from the whole body DNA extract of one female beetle from the laboratory stock. The raw data constitutes about 31 GB nanopore long read sequence and 125 GB illumina pair-end short read illumina sequence. Raw sequence data is quality assessed using [FastQC](https://github.com/s-andrews/FastQC) for short reads and [NanoPlot](https://github.com/wdecoster/NanoPlot) for long read data.

```bash
# Retrieving long-reads specific to Wolbachia
minimap2 -t $threads -a wolb.scaf.fa lr.corrected.fa.gz > wolb.scaf.sam
cat wolb.scaf.sam | samtools view -buS - | samtools sort - -o wolb.scaf.sorted.bam 
samtools view -b -F 4 wolb.scaf.sorted.bam > wolb.scaf.mapped.bam 
samtools fasta wolb.scaf.mapped.bam > wolb.scaf.mapped.fa
```
```bash
# Assemblying Wolbachia genome from the long-reads with 'Flye' (F-assembly)
flye --meta --nano-corr wolb.scaf.mapped.fa --out-dir assembly/flye --threads $threads 
cp assembly.fasta wolb.flye.fa
```
> The resulting <.html> files from FastQC and NanoPlot are manually inspected for sequence metadata and read quality assessment.

In Nanopore sequencing platform, the DNA sequences are read through the changes in voltage. As a result, the base errors are randomly distributed through out the entire length of reads. Hence, there is no end trimming of Nanopore long reads. However, quality control is a default process during the Guppy base calling, which is mostly carried out by the NGS facility or the service provider, that filters out the entire reads with average Phred quality score. In the current data, the minimum Phred quality score is 9.

However, the illumina short read data is quality trimmed as follows using [FastP](https://github.com/OpenGene/fastp).
```bash
# Running FastP
fastp -i ~/path/to/dir/in/sr_R1.fastq.gz -o ~/path/to/dir/out/sr.corrected.R1.fq.gz \
 -I ~/path/to/dir/in/sr_r2.fastq.gz -O ~/path/to/dir/out/sr.corrected.R2.fq.gz \
 --detect_adapter_for_pe --trim_poly_g -5 -3 -l 51 -j ~/path/to/dir/out/$prefix.json \ 
 -h ~/path/to/dir/out/$prefix.html -w $threads
```
