# Assembly and Annotation

The input data in this pipeline includes short-read <sr_R1.fastq.gz, sr_R2.fastq.gz> and error-corrected long-read <lr_corrected.fastq.gz> of the whole genome sequence data of the host beetle species *Zygogramma bicolorata*. During the genome assembly of the host beetle, 11 contigs were identified as of *Wolbachia* origin. These 11 *Wolbachia*-specific contigs are in the file wolb.scaf.fa. 

In this pipeline, the 11 *Wolbachia*-specific contigs were used to retrieve *Wolbachia*-specific long reads and short-reads from the whole genome data of the beetle species. These reads were then used to assemble the *Wolbachia* genome. Two methods were used for bacterial genome assembly, [Flye](https://github.com/mikolmogorov/Flye) and [B-assembler](https://github.com/ChongLab/B-assembler).

```bash
# Retrieving long-reads specific to Wolbachia
minimap2 -t $threads -a wolb.scaf.fa lr.corrected.fa.gz > wolb.scaf.sam
cat wolb.scaf.sam | samtools view -buS - | samtools sort - -o wolb.scaf.sorted.bam 
samtools view -b -F 4 wolb.scaf.sorted.bam > wolb.scaf.mapped.bam 
samtools fasta wolb.scaf.mapped.bam > wolb.lr.fa
```
> The resulting <wolb.lr.fa> contains *Wolbachia*-specific long reads.
```bash
# Wolbachia genome assembly from the long-reads with 'Flye'
flye --meta --nano-corr wolb.lr.fa --out-dir assembly/flye --threads $threads 
cp assembly.fasta wolb.flye.fa
```
> Flye-based assembly is in <wolb.flye.fa>.
```bash
# Using the Flye-assembly to retrieve Wolbachia specific short-reads
bwa-mem2 index -p wolb.flye wolb.flye.fa
bwa-mem2 mem -t 90 -P wolb.flye sr_R1.fastq.gz sr_R2.fastq.gz | samtools view -@ 10 -buS - | samtools sort -@ 10 - -o wolb.sr.sorted.bam
samtools view -b -@ 110 -F 4 wolb.sr.sorted.bam > wolb.sr.filter.bam
samtools fastq -@ 110 wolb.sr.filter.bam -1 wolb_R1.fastq.gz -2 wolb_R2.fastq.gz -0 /dev/null -n
```
> *Wolbachia*-specific short-reads are in <wolb_R1.fastq.gz, wolb_R2.fastq.gz>.
```bash
# Polca polishing the Flye-assembly with Wolbachia-specific short-reads
polca.sh -a wolb.flye.fa -r 'wolb_R1.fastq.gz -2 wolb_R2.fastq.gz' -t 110 -m 2G
cp wolb.flye.fa.Polca*.fa wolb.flye.corrected.fa
```
> Final corrected Flye-based assembly is in <wolb.flye.corrected.fa>.
```bash
# Assemblying Wolbachia genome from the long-reads with 'B-assembler'.
source activate B-assembler 
run_B-assembler.sh <numCPUs> LongReadOnly <PWD>

# Polca polishing the B-assembly with Wolbachia specific short-reads
polca.sh -a wolb.bass.fa -r 'wolb_R1.fastq.gz -2 wolb_R2.fastq.gz' -t 60 -m 2G
cp wolb.bass.fasta.Polca*.fa wolb.bass.corrected.fa 
```
> Final corrected B-assembler assembly is in <wolb.bass.corrected.fa>.

Comparison of output from Flye and B-assembler.
```bash
tblastn -query wolb.flye.corrected.fa -subject wolb.bass.corrected.fa -out file -outfmt 7
```

Assembly evaluation with [checkM](https://github.com/OpenGene/fastp) and [BUSCO](https://github.com/OpenGene/fastp).
```bash
# CheckM: rickettsiales
checkm data setRoot ~/B_assembler/checkm_data/ 
checkm lineage_wf -x fa -t 16 ~/B_assembler/bins/ ~/B_assembler/checkM/
# BUSCO: rickettsiales_odb11
busco -i wolb.bass.corrected.fa -m genome -c 60 --force -l rickettsiales_odb10 --out wolb_busco --offline --download_path ~/busco_downloads
```
Various online tools can now be used to annotate the bacterial genome: (i) OriC localization and annotation using [OriFinder](https://tubic.org/Ori-Finder/), (ii) whole genome annotation using [PGAP](https://github.com/ncbi/pgap). Results from PGAP annotation can be evaluated for BUSCO completeness analysis using [gVolante](https://gvolante.riken.jp/) web server. Orthologus and functional analyses of coding sequences can be achieved through [COG](https://www.ncbi.nlm.nih.gov/research/cog/) and [KEGG](https://www.genome.jp/kegg/) databases and the online tools available there.

Annotation of repeatative elements. [digIS](https://github.com/janka2012/digIS) used to map the IS elements. [P-assembler](https://github.com/gbouras13/plassembler) used to detect the presence of plasmid-like elements by using the reads that were not mapped to the host beetle genome, hence includes *Wolbachia*-specific reads and other associated reads. Additionally, two reference plasmid sequences were used to identify the presence of plasmid-associated elements in assembled *Wolbachia* genome through blast search. Annotation of PhageWO elements mostly included use of [Phaster](https://phaster.ca/) online server and compilation of data from PGAP annotation and manual walk-out method and correction.
```bash
# IS elements
python digIS/digIS_search.py -i wolb.bass.corrected.fa -o digIS

# Plasmid assembly in hybrid mode
plassembler run -d ./DB -l lr.unmapped.fastq -o ./output -1 sr.unmapped_R1.fastq -2 sr.unmapped_R2.fastq -c 1400000 --threads 64
tblastn -query pWALBA1.aa -db wol -out file -evalue 0.05 -outfmt 7
tblastn -query pWALBA2.aa -db wol -out file -evalue 0.05 -outfmt 7
```
phageWO classification based on serine recombinase (SR) sequence. Total 17 homologs were compiled to prepare a dataset <serine_recom_17cds.fa>, which was analyzed for Phylogeny reconstruction by using [mafft](https://mafft.cbrc.jp/alignment/server/index.html) and [iqtree](https://github.com/iqtree/iqtree2).
```bash
cat serine_recomb_cds.fasta | seqkit grep -p "CP149530_WKH12_03425_WOBic4" | seqkit translate -T 11 --frame 6     #all frames have pseudo regions
mafft --auto --thread 8 serine_recomb_17cds.fa > sr_cds_aln.fa
iqtree2 -s sr_cds_aln.fa -m MFP -T 10 -B 1000
```
