# Assembly and Annotation

The input data in this pipeline includes short-read (sr_R1.fastq.gz, sr_R2.fastq.gz) and error-corrected long-read (lr_corrected.fastq.gz) of the whole genome sequence data of the host beetle species *Zygogramma bicolorata*. During the genome assembly of the host beetle, 11 contigs were identified as of *Wolbachia* origin. These 11 *Wolbachia*-specific contigs are in the file wolb.scaf.fa. 
In this pipeline, the 11 *Wolbachia*-specific contigs were used to retrieve *Wolbachia*-specific long reads and short-reads from the whole genome data of the beetle species. These reads were then used to assemble the *Wolbachia* genome.

```bash
# Retrieving long-reads specific to Wolbachia
minimap2 -t $threads -a wolb.scaf.fa lr.corrected.fa.gz > wolb.scaf.sam
cat wolb.scaf.sam | samtools view -buS - | samtools sort - -o wolb.scaf.sorted.bam 
samtools view -b -F 4 wolb.scaf.sorted.bam > wolb.scaf.mapped.bam 
samtools fasta wolb.scaf.mapped.bam > wolb.lr.fa
```
> The resulting <wolb.lr.fa> contains *Wolbachia*-specific long reads.
```bash
# Assemblying Wolbachia genome from the long-reads with 'Flye' (F-assembly)
flye --meta --nano-corr wolb.lr.fa --out-dir assembly/flye --threads $threads 
cp assembly.fasta wolb.flye.fa
```
> Flye-based assembly is in <wolb.flye.fa>
```bash
# Using the F-assembly to retrieve Wolbachia specific short-reads
bwa-mem2 index -p wolb.flye wolb.flye.fa
bwa-mem2 mem -t 90 -P wolb.flye sr_R1.fastq.gz sr_R2.fastq.gz | samtools view -@ 10 -buS - | samtools sort -@ 10 - -o wolb.sr.sorted.bam
samtools view -b -@ 110 -F 4 wolb.sr.sorted.bam > wolb.sr.filter.bam
samtools fastq -@ 110 wolb.sr.filter.bam -1 wolb_R1.fastq.gz -2 wolb_R2.fastq.gz -0 /dev/null -n

# Polca polishing the F-assembly with Wolbachia specific short-reads (not sure if executed)
polca.sh -a wolb.flye.fa -r 'wolb_R1.fastq.gz -2 wolb_R2.fastq.gz' -t 110 -m 2G
cp wolb.flye.fa.Polca*.fa wolb.flye.corrected.fa
```
```bash
# Assemblying Wolbachia genome from the long-reads with 'B-assembler' (B-assembly)
source activate B-assembler 
run_B-assembler.sh 110 LongReadOnly ~/Wolbachia/B_assembler

# Polca polishing the B-assembly with Wolbachia specific short-reads
polca.sh -a wolb.bass.fa -r 'wolb_R1.fastq.gz -2 wolb_R2.fastq.gz' -t 60 -m 2G
cp wolb.bass.fasta.Polca*.fa wolb.bass.corrected.fa 

# Comparing F-assembly and B-assembly
tblastn -query wolb.flye.corrected.fa -subject wolb.bass.corrected.fa -out file -outfmt 7
```

> The resulting <.html> files from FastQC and NanoPlot are manually inspected for sequence metadata and read quality assessment.

Assembly evaluation with [checkM](https://github.com/OpenGene/fastp) and [BUSCO](https://github.com/OpenGene/fastp).

```bash
# CheckM: rickettsiales
checkm data setRoot ~/Wolbachia/B_assembler/checkm_data/ 
checkm lineage_wf -x fa -t 16 ~/Wolbachia/B_assembler/bins/ ~/Wolbachia/B_assembler/checkM/
# BUSCO: rickettsiales_odb11, proteobacteria_odb11
busco -i polca/wolb.bass.corrected.fa -m genome -c 60 --force -l rickettsiales_odb10 --out wolbachia_busco --offline --download_path ~/Wolbachia/busco/busco_downloads
```
Various online tools can be used to annotate the bacterial genome: (i) OriC localization and annotation using [OriFinder](https://tubic.org/Ori-Finder/), (ii) IS element identification with [digIS](https://github.com/janka2012/digIS), (iii) whole genome annotation using [PGAP](https://github.com/ncbi/pgap). Results from PGAP annotation can be evaluated for BUSCO completeness analysis using [gVolante](https://gvolante.riken.jp/) web server. Orthologus and functional analyses of coding sequences can be achieved through [COG](https://www.ncbi.nlm.nih.gov/research/cog/) and [KEGG](https://www.genome.jp/kegg/) databases and the online tools available there.

Annotation of repeat elements
```bash
# IS elements
python ~/digIS/digIS_search.py -i wolb.bass.corrected.fa -o digIS

# Prophage sequences (WO): PHASTER; Pharokka; comparative blast
> Phaster analysis using online server
> Pharokka analysis in Galaxy server

# Plasmid assembly in hybrid mode
plassembler run -d ./DB -l BP23.mini.lr.unmapped.fastq -o ./output -1 BP23.sr.unmapped_R1.fastq -2 BP23.sr.unmapped_R2.fastq -c 1400000 --threads 64
tblastn -query pWALBA1.aa -db wol -out file -evalue 0.05 -outfmt 7
tblastn -query pWALBA2.aa -db wol -out file -evalue 0.05 -outfmt 7

# Serine recombinase classification
cat serine_recomb_cds.fasta | seqkit grep -p "CP149530_WKH12_03425_WOBic4" | seqkit translate -T 11 --frame 6     #all frames have pseudo regions
mafft --auto --thread 8 serine_recomb_17cds.fasta > sr_cds_aln.fasta     #FFT-NS-i model is faster while L-INS-i more accurate
wd=/mnt/c/Users/SAHOO/Documents/Sahoo/Projects/workdir/dumpack/fasttree
$wd/FastTree -nt -gtr -gamma sr_cds_aln.fasta > sr_cds_ftree.nwk
> gamma option rescale the branch lengths and compute a Gamma20-based likelihood.
> FastTree estimate the reliability of each split in the tree by using the Shimodaira-Hasegawa test on the three alternate topologies around that split.
iqtree2 -s sr_cds_aln.fasta -m MFP -T 10 -B 1000

```


