# Genome comparisons

Here, we compiled a dataset of 242 *Wolbachia* genomes that included wBic plus previously compiled 199 genomes and 42 additional genomes from Genbank. This genome set is in <wolb242.fa>, and was used for comparative analyses. In the analyses, Prokka used to predict the CDS regions across genomes, which was then used to analyse the shared orthoroups. 
```bash
# Phylogeny reconstruction
mkdir wolb242_proteome wolb242_output wolb242_gene
for i in `ls wolb242`; do prokka --outdir ./prokka_wolb242/$i --prefix $i --locustag $i --prefix $i --genus wolbachia --kingdom Bacteria --noanno --cpus 20 wolb242/$i; done
mkdir wolb242_proteome
for i in `ls wolb242`; do cp prokka_wolb242/$i/$i.faa wolb242_proteome; done  # Protein file is moved to a single directory 
# Orthofinder is run till orthogroup prediction 
orthofinder -t 20 -o /mnt/bioinfo/Mexican_beelte/Wolbachia/OrthoFinder/run/Orthofinder_results_og -f ./wolb242_proteome/  -og
Orthogroups.GeneCount.tsv:: =COUNTIF(B2:GS2,"1") & =MAX(B2:GS2) > 95.orthogroups.txt
# for finding the duplicate and removing it 
for i in `ls 95_orthogroups_edited/`; do cat 95_orthogroups_edited/$i | grep ">" | sort | uniq -c | grep -v "1 " | sed 's/.*>//g' > temp/$i.dup.txt; seqkit grep -v -f temp/$i.dup.txt 95_orthogroups_edited/$i > 95_orthogroups_final/$i; done
for i in `ls 95_orthogroups_final/*`; do cat $i | grep ">" | sort | uniq -c | grep "1 " | wc ; done
for i in `ls ./95_orthogroups_final`; do mafft --auto --thread 16 ./95_orthogroups_final/$i > ./mafft/$i.aln; done
for i in `ls ./mafft`; do trimal -automated1 -in ./mafft/$i -out ./trimal/$i; done #(WARNING: Removing sequence 'GCA_902636395_F' composed only by gaps)
cd /home/sahoork/dumpack/phyutility
./phyutility -concat -aa -in ./trimal/* -out ./OG613_supermatrix.nexus
iqtree2 -s ~/Wolbachia/Orthology/OG613_supermatrix.nexus -T 72 -m LG+G4 -B 1000 --prefix ~/Wolbachia/WorkDir/output/iqtree_B1000/result/OG613
```

```bash
#gc content comparison (126 circular genomes)
seqkit fx2tab --infile-list ../circular_genomes -n -g -G -l -i > cir_genomes.gc
seqkit stats --infile-list ../circular_genomes  > cir_genomes.stats
```

```bash
## Comparative analyses ##
# Genome reorientation
conda activate dnaapler
dnaapler chromosome -i data/wBic.fasta -o wbic -p wBic -t 8
dnaapler chromosome -i data/wTae.fasta -o wtae -p wTae -t 8
dnaapler chromosome -i data/wCon.fasta -o wcon -p wCon -t 8
cp wbic/wBic_reoriented.fasta wtae/wTae_reoriented.fasta wcon/wCon_reoriented.fasta ./dnaapler

#FastANI
mkdir out
./fastANI -q dnaapler/wTae_reoriented.fasta -r dnaapler/wBic_reoriented.fasta --visualize -o out/ani_taebic_reor.out
./fastANI -q dnaapler/wCon_reoriented.fasta -r dnaapler/wBic_reoriented.fasta --visualize -o out/ani_conbic_reor.out
./fastANI -q dnaapler/wTae_reoriented.fasta -r dnaapler/wCon_reoriented.fasta --visualize -o out/ani_taecon_reor.out
cat out/ani_taebic_reor.out >> out/summary.txt
cat out/ani_conbic_reor.out >> out/summary.txt
cat out/ani_taecon_reor.out >> out/summary.txt

#pygenome plot
conda activate pygenomeviz
python scripts/visualize.py dnaapler/wTae_reoriented.fasta dnaapler/wBic_reoriented.fasta out/ani_taebic_reor.out.visual out/taebic.png --cmap viridis --link_color red --curve
python scripts/visualize.py dnaapler/wCon_reoriented.fasta dnaapler/wBic_reoriented.fasta out/ani_conbic_reor.out.visual out/conbic.png --cmap viridis --link_color red --curve
python scripts/visualize.py dnaapler/wTae_reoriented.fasta dnaapler/wCon_reoriented.fasta out/ani_taecon_reor.out.visual out/taecon.png --cmap viridis --link_color red --curve

#ortho comparison
>using OrthoVenn online server
#digital DDH
>calculated using online server GGDC <ggdc.dsmz.de>
#dot plots
>plotted online with D-genies https://dgenies.toulouse.inra.fr/
#Artemis Comparison tools (not used)
perl ./scripts/abacas.1.03.pl -r data/wBic.fasta -q data/wTae.fasta -p nucmer
```
