# Identification and characterization of *wmk*

Input files for this analyses are (i) <wBic.protein.fa>: the amino acid sequences of predicted CDS regions of wBic genome, (ii) <ref.protein-wmk.fa>: Wmk amino acid sequences from three other *Wolbachia* strains.
```bash
# Wmk identification
blastp -query ref.rotein-wmk.fa -subject wBic.protein.fa -out wmk-blastp.out -outfmt 7
mmseqs easy-search ref.protein-wmk.fa wBic.protein.fa wmk-mmseq.html tmp -a --format-mode 3
mmseqs easy-search ref.protein-wmk.fa wBic.protein.fa wmk-mmseq.out tmp -a --format-mode 4
mmseqs easy-rbh wmk-candidates.fa ref.protein-wmk.fasta wmk-final.out tmp --format-mode 4
```
```
# Homology search
blastp -query dataset1.fasta -subject dataset1.fasta -out dataset1-blastp.out -outfmt 7
mmseqs easy-search wmel-wmk.fasta dataset1.fasta dataset1-mmseq.html tmp -a --format-mode 3
mmseqs easy-search wmel-wmk.fasta dataset1.fasta dataset1-mmseq.out tmp --format-mode 4
```
>> Homology search plotted in R


# MSA and conservation calculation
>>MSA conducted in T-Coffee server using M-Coffee #(mafft alignment was same as M-coffee)
conda activate py27 && cd /mnt/c/Users/SAHOO/Documents/Sahoo/Projects/workdir/dumpack/conservation_code/
python ./score_conservation.py -l FALSE -s shannon_entropy -w 0 -o ./wmk/dataset3_conserv.txt ./wmk/dataset3_aln.fasta
cat ./wmk/dataset3_conserv.txt | sed '/^#/d' | sed 's/-1000/0/g' > ./wmk/dataset3_conserv.out
>> Plotted in R

#Wmk phylogeny (sequence based)
mafft --auto --thread 8 ../dataset3.fasta > dataset3_aln_mafft.fasta     #FFT-NS-i model is faster while L-INS-i more accurate
iqtree2 -s dataset3_aln_mafft.fasta -m MFP -T 10 -B 1000
