# Identification and characterization of *wmk*

Input files for this analyses are (i) <wBic.protein.fa>: the amino acid sequences of predicted CDS regions of wBic genome, (ii) <ref.protein-wmk.fa>: Wmk amino acid sequences from three other *Wolbachia* strains. The reference Wmk sequences were used to identify the putative homolog in wBic using [MMseqs2](https://github.com/soedinglab/MMseqs2).
```bash
# Wmk identification
blastp -query ref.rotein-wmk.fa -subject wBic.protein.fa -out wmk-blastp.out -outfmt 7
mmseqs easy-search ref.protein-wmk.fa wBic.protein.fa wmk-mmseq.html tmp -a --format-mode 3
mmseqs easy-search ref.protein-wmk.fa wBic.protein.fa wmk-mmseq.out tmp -a --format-mode 4
mmseqs easy-rbh wmk-candidates.fa ref.protein-wmk.fasta wmk-final.out tmp --format-mode 4
```
27 Wmk homologs from 23 *Wolbachia* strains (in the file <dataset1.fa>) are compared to check sequence-level divergence against the reference Wmk from *Wolbachia* infecting *Drosophila melanogaster*.
```
mmseqs easy-search wMel-wmk.fa dataset1.fa dataset1-mmseq.html tmp -a --format-mode 3
mmseqs easy-search wMel-wmk.fa dataset1.fa dataset1-mmseq.out tmp --format-mode 4
```
A subset of <dataset1.fa> containing 19 Wmk homologs from 13 *Wolbachia* strains (in the file <dataset3.fa>) is used for subsequent analyses. Multiple sequence analysis of this new dataset was conducted in [M-Coffee](https://tcoffee.crg.eu/) web server, and the alignment can be visualized using another online tool [Alignmentviewer](https://fast.alignmentviewer.org/). The aligned file was used to calculate residue wise conservation score using the script available at [PRCP](https://compbio.cs.princeton.edu/conservation/index.html).
```
# Conservation score calculation
conda activate py27 && cd ~/conservation_code/
python ./score_conservation.py -l FALSE -s shannon_entropy -w 0 -o ./dataset3_conserv.txt ./dataset3_aln.fasta
cat dataset3_conserv.txt | sed '/^#/d' | sed 's/-1000/0/g' > dataset3_conserv.out
```
>> The output was plotted using R.

Wmk homologs in <dataset3.fa> were aligned with [mafft](https://mafft.cbrc.jp/alignment/server/index.html) and analysed for phylogenetic relationships using [iqtree](https://github.com/iqtree/iqtree2).
```
mafft --auto --thread 8 ../dataset3.fa > dataset3_aln_mafft.fasta     # FFT-NS-i model is faster while L-INS-i more accurate
iqtree2 -s dataset3_aln_mafft.fasta -m MFP -T 10 -B 1000
```
